src/Learners/Core/Core.csproj

```diff
@@ -32,4 +32,8 @@
     <Compile Include="..\..\Shared\SharedAssemblyFileVersion.cs" />
     <Compile Include="..\..\Shared\SharedAssemblyInfo.cs" />
   </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
```

Comment:
Newtonsoft doesn't handle everything, and many of the types cannot currently be serialised by Newtonsoft. I'm not sure we want to redesign everything to be serializable by Newtonsoft. I think the manual ForwardCompatible-scheme is a better and future-proof model if a large redesign is desired.

Also note using Formatters where possible allows backward compatibility because people can provide their own BinaryFormatter if they choose, even if only as a bridge to recover old data.

Reply:
Then why are we taking a dependency on Newtonsoft?  The way I see it, it's either one or the other.  

BinaryFormatter is not a long term storage format.  We only support backward compatibility through the methods that explicitly say it.